src/log/log.c

```diff
@@ -2452,24 +2462,12 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
 			WT_STAT_CONN_INCR(session, log_scan_rereads);
 		}
 		/*
-		 * We read in the record, now make some checks.
-		 * These checks are conservative, if they fail, there
-		 * is definitely corruption.
+		 * We read in the record, now verify the checksum.  A failed
+		 * checksum does not imply corruption, it may be the result
+		 * of a partial write.
 		 */
 		buf->size = reclen;
 		logrec = (WT_LOG_RECORD *)buf->mem;
-		WT_ERR(__log_record_verify(session, log_fh, rd_lsn.l.offset,
-		    logrec, &corrupt));
-		if (corrupt) {
-			need_salvage = true;
-			WT_ERR(__log_salvageable_error(session, log_fh->name,
-			    "", rd_lsn.l.offset));
-		}
-
```

Comment:
While this all looks fine to me, to be clear, the original code was verifying the header on every record read in. Now this fix will only verify the header when there is a checksum failure.

Reply:
That's correct.  I needed to reorder the code anyway (can't byte swap before computing the checksum), and I realized that the reason for the header checks is a guard against corruption.  If there is no checksum failure, there isn't corruption. Given that the header check now needed to be more involved (copy the header, and byte swap it), it seemed silly to make the check when we didn't really need it.